docs/src/concepts/data_management.md

```diff
@@ -0,0 +1,61 @@
+# Data management
+
+This section covers concepts related to managing your data over time in LanceDB.
+
+## A primer on Lance
+
+Because LanceDB is built on top of the [Lance](https://lancedb.github.io/lance/) data format, it helps to understand some of its core ideas. Just like Apache Arrow, Lance is a fast columnar data format, but it has the added benefit of being versionable, query and train ML models on. Lance is designed to be used with simple and complex data types, like tabular data, images, videos audio, 3D point clouds (which are deeply nested) and more.
+
+The following concepts are important to keep in mind:
+
+- Data storage is columnar and is interoperable with other columnar formats (such as Parquet) via Arrow
+- Data is divided into fragments that represent a subset of the data
+- Data is versioned, with each insert operation creating a new version of the dataset and an update to the manifest that tracks versions via metadata
+
+!!! note
+    Understanding what causes a new version to be created is important to keep your manifests clean and file sizes reasonable. In Lance, each insert operation creates a new version of the dataset. If you batch-insert 100 rows via a single command, a single version is created. If you insert 100 rows one at a time, it will create **100 versions**. This is important to keep in mind when designing your data ingestion pipeline.
```

Comment:
@changhiskhan and @wjones127, I've added this file in the `Concepts` section of the docs to describe our data management concepts like compaction and re-indexing. This particular example I state was to illustrate how versioning works -- does this look correct? Feel free to suggest changes in this bit or any other part of this file when you can. Thanks!